test/test_overrides.py

```diff
@@ -290,6 +310,7 @@ def generate_tensor_like_torch_implementations():
     testing_overrides = get_testing_overrides()
     for namespace, funcs in get_overridable_functions().items():
         for func in funcs:
+            HANDLED_FUNCTIONS_NAMESPACES[func] = namespace
```

Comment:
This seems a bit smelly to me.

The original intention of `HANDLED_FUNCTIONS_TENSOR_LIKE` was that it was to show one idiomatic way to modularly define torch functions overrides (rather than writing a giant if statement inside your `__torch_function__`. That's why it gets mutated in `implements_tensor_like` decorator: you use this decorator, and then your function implementation gets schlorped into the table. Now, it *turns out* that `implements_tensor_like` is only ever called from a metaprogramming loop (non-idiomatic), but at least the top level structure makes sense.

With the changes here, we end up with some code structure that doesn't really make sense if you were planning on just overwriting a few methods (rather than metaprogramming). More concretely, I would have expected some sort of function like `implements_tensor_method` which you hypothetically could have used to program methods one-by-one.

Reply:
Or maybe, this doesn't make that much sense (because the point of `__torch_function__` on methods is to let you universally override things, and if you only needed to override one method, you should have just defined that method directly), but I don't *think* that's true in our latest design, and even if it was true, we should make it clear that this is operating differently than `implements_tensor_like`

Reply:
Both are possible, but if you want to use the implementation of PyTorch functions/methods, then you must use `super().__torch_function__(func, ...)`.

Reply:
Action item: Add the double inheritance hierarchy diagram to the docs.